neutron/services/loadbalancer/agent/agent_manager.py

```diff
@@ -332,3 +332,24 @@ def agent_updated(self, context, payload):
                              pool_id)
                     self._destroy_pool(pool_id)
             LOG.info(_("Agent_updated by server side %s!"), payload)
+
+    def add_pool(self, context, pool, driver_name):
```

Comment:
add_pool这里没有关注底层的create_pool, 是自己底层新加的add_pool对应的函数 已经处理了create pool的流程。。。

2016-07-13 14:19 GMT+08:00 huntxu <notifications@github.com>:

> In neutron/services/loadbalancer/agent/agent_manager.py
> <https://github.com/eayunstack/neutron/pull/39#discussion_r70569334>:
>
> > @@ -332,3 +332,24 @@ def agent_updated(self, context, payload):
> >                               pool_id)
> >                      self._destroy_pool(pool_id)
> >              LOG.info(_("Agent_updated by server side %s!"), payload)
> > +
> > +    def add_pool(self, context, pool, driver_name):
>
>
> 我之前也注意是不是要先调用create_pool然后update_pool才能正常工作，但我看你add_pool这里也没有关注底层的create_pool方面的东西整个的流程最终在agent那边是和调用update_pool一个效果，所以大概直接发一个update_pool也是可行的？还是我有哪里没注意到？
>
> 不过从另一方面考虑整个的完整性的或者不破坏其他driver的话，的确应该先create_pool再update_pool的
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eayunstack/neutron/pull/39/files/609dd87c1c005386d538a761bbb1b4b09842b8c4#r70569334>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGUT0SbQgm-wh0786V5HNUuB2tGZCPHqks5qVINxgaJpZM4JJO13>
> .
>